interactive_engine/executor/ir/runtime/src/process/entry.rs

```diff
@@ -0,0 +1,619 @@
+                .downcast_ref::<Intersection>()
+                .hash(state),
+            EntryDataType::Collection => self
+                .as_any_ref()
+                .downcast_ref::<CollectionEntry>()
+                .hash(state),
+        }
+    }
+}
+
+// demanded when need to key the entry; and order the entry;
+impl PartialEq for DynEntry {
+    fn eq(&self, other: &Self) -> bool {
+        if (self.get_type()).eq(&other.get_type()) {
+            match self.get_type() {
+                EntryDataType::Vid | EntryDataType::V => self.as_vid().eq(&other.as_vid()),
+                EntryDataType::E => self.as_edge().eq(&other.as_edge()),
+                EntryDataType::P => self.as_graph_path().eq(&other.as_graph_path()),
+                EntryDataType::Obj => self.as_object().eq(&other.as_object()),
+                EntryDataType::Intersect => self
+                    .as_any_ref()
+                    .downcast_ref::<Intersection>()
+                    .eq(&other
+                        .as_any_ref()
+                        .downcast_ref::<Intersection>()),
+                EntryDataType::Collection => self
+                    .as_any_ref()
+                    .downcast_ref::<CollectionEntry>()
+                    .eq(&other
+                        .as_any_ref()
+                        .downcast_ref::<CollectionEntry>()),
+            }
+        } else {
+            false
+        }
+    }
+}
+
+// demanded when need to  order the entry;
+impl PartialOrd for DynEntry {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        if (self.get_type()).eq(&other.get_type()) {
+            match self.get_type() {
+                EntryDataType::Vid | EntryDataType::V => self.as_vid().partial_cmp(&other.as_vid()),
+                EntryDataType::E => self.as_edge().partial_cmp(&other.as_edge()),
+                EntryDataType::P => self
+                    .as_graph_path()
+                    .partial_cmp(&other.as_graph_path()),
+                EntryDataType::Obj => self.as_object().partial_cmp(&other.as_object()),
+                EntryDataType::Intersect => self
+                    .as_any_ref()
+                    .downcast_ref::<Intersection>()
+                    .partial_cmp(
+                        &other
+                            .as_any_ref()
+                            .downcast_ref::<Intersection>(),
+                    ),
+                EntryDataType::Collection => self
+                    .as_any_ref()
+                    .downcast_ref::<CollectionEntry>()
+                    .partial_cmp(
+                        &other
+                            .as_any_ref()
+                            .downcast_ref::<CollectionEntry>(),
+                    ),
+            }
+        } else {
+            None
+        }
+    }
+}
+
+// demanded when need to group (ToSet) the entry;
+impl Eq for DynEntry {}
+
+// demanded when need to group (ToSum) the entry;
+impl std::ops::Add for DynEntry {
```

Comment:
Feels like we should not implement `Add` for entry.

Reply:
Done. Move this logic into `Accumulator`.